Microsoft.Toolkit.Uwp.UI.Controls/Eyedropper/Eyedropper.cs

```diff
@@ -0,0 +1,251 @@
+            _targetGrid.PointerPressed += TargetGrid_PointerPressed;
+            _targetGrid.PointerMoved += TargetGrid_PointerMoved;
+            _targetGrid.PointerReleased += TargetGrid_PointerReleased;
+        }
+
+        private void UnhookEvents()
+        {
+            Unloaded -= Eyedropper_Unloaded;
+            Window.Current.SizeChanged -= Window_SizeChanged;
+            DisplayInformation.GetForCurrentView().DpiChanged -= Eyedropper_DpiChanged;
+            if (_targetGrid != null)
+            {
+                _targetGrid.PointerEntered -= TargetGrid_PointerEntered;
+                _targetGrid.PointerExited -= TargetGrid_PointerExited;
+                _targetGrid.PointerPressed -= TargetGrid_PointerPressed;
+                _targetGrid.PointerMoved -= TargetGrid_PointerMoved;
+                _targetGrid.PointerReleased -= TargetGrid_PointerReleased;
+            }
+        }
+
+        private void Eyedropper_Loaded(object sender, RoutedEventArgs e)
+        {
+            _lazyTask?.Invoke();
+            _lazyTask = null;
+        }
+
+        private void TargetGrid_PointerExited(object sender, PointerRoutedEventArgs e)
+        {
+            Window.Current.CoreWindow.PointerCursor = DefaultCursor;
+            if (_pointerId != null)
+            {
+                _pointerId = null;
+                Opacity = 0;
+            }
+        }
+
+        private void TargetGrid_PointerEntered(object sender, PointerRoutedEventArgs e)
+        {
+            Window.Current.CoreWindow.PointerCursor = MoveCursor;
+        }
+
+        private async void Eyedropper_DpiChanged(DisplayInformation sender, object args)
+        {
+            await UpdateAppScreenshotAsync();
+        }
+
+        private void TargetGrid_PointerReleased(object sender, PointerRoutedEventArgs e)
+        {
+            var pointer = e.Pointer;
+            if (pointer.PointerId == _pointerId)
+            {
+                var point = e.GetCurrentPoint(_rootGrid);
+                UpdateEyedropper(point.Position);
+                PickCompleted?.Invoke(this, EventArgs.Empty);
+                _pointerId = null;
+                if (!_taskSource.Task.IsCanceled)
+                {
+                    _taskSource.SetResult(Color);
+                }
+            }
+        }
+
+        private void TargetGrid_PointerMoved(object sender, PointerRoutedEventArgs e)
+        {
+            var pointer = e.Pointer;
+            if (pointer.PointerId == _pointerId)
+            {
+                var point = e.GetCurrentPoint(_rootGrid);
+                UpdateEyedropper(point.Position);
+            }
+        }
+
+        private async void TargetGrid_PointerPressed(object sender, PointerRoutedEventArgs e)
+        {
+            _pointerId = e.Pointer.PointerId;
+            PickStarted?.Invoke(this, EventArgs.Empty);
+            await UpdateAppScreenshotAsync();
+            var point = e.GetCurrentPoint(_rootGrid);
```

Comment:
I've found that if you activate the eyedropper and quickly tap the screen there is a chance the app might crash in this line, it looks like the await causes a race condition and if the pointer is released before UpdateAppScreenshotAsync finishes then GetCurrentPoint will fail. I'd suggest swapping these two lines: 
```csharp
var point = e.GetCurrentPoint(_rootGrid); 
await UpdateAppScreenshotAsync();
```

Reply:
@arcadiogarcia Thanks, but I tried it many times and didn't reproduce the bug. I didn't test it on a real touch device, maybe it only happens on touch devices?
Anyway, I think your suggestion makes sense, so I modified the code.

Reply:
Yeah, for some reason I can only reproduce the issue with touch, pen and mouse never break, I'm not quite sure why tbh.